.github/workflows/ci.yml

```
@@ -559,37 +559,37 @@ jobs:
         set +e; source /opt/intel/oneapi/setvars.sh; set -e
         cmake --build build-11 --target test_cmake_build
 
-    - name: Configure C++14
+    - name: Configure C++17
       run: |
         set +e; source /opt/intel/oneapi/setvars.sh; set -e
-        cmake -S . -B build-14     \
+        cmake -S . -B build-17     \
         -DPYBIND11_WERROR=ON    \
```

Maybe we can then compile without `-Werror-all` on `icpc` for now:
```suggestion
        -DPYBIND11_WERROR=OFF   \
```